File: HolyChildhood/ClientApp/src/app/shared/models/page-content.ts
import {Page} from './page';
import {Calendar} from './calendar';
import {Pdf} from './file';

export interface PageContent {
    id: number;
    contentType: string;
    hasTitle: boolean;
    title: string;
    editing: boolean;
    index: number;
    pageId: number;
    page: Page;
    textContent: TextContent;
    tabContent: TabContent;
    calendarContent: CalendarContent;
    fileContent: FileContent;
}

export interface TextContent {
    id: number;
    content: string;
}

export interface TabContent {
    id: number;
    tabs: Tab[];
}

export interface Tab {
    id: number;
    title: string;
    index: number;
    tabContentId: number;
    textContent: TextContent;
}

export interface CalendarContent {
    id: number;
    calendarId: number;
    calendar: Calendar;
}

export interface FileContent {
    id: number;
    fileType: string;
    files: Pdf[];
}

export interface TextContentBackup {
    id: number;
    creationDate: Date;
    displayDate: string;
    content: string;
    textContentId: number;
    textContent: TextContent;
}

File: HolyChildhood/Models/PageContent.cs
using System;
using System.Collections.Generic;

namespace HolyChildhood.Models
{
    public class PageContent
    {
        public int Id { get; set; }
        public string ContentType { get; set; }
        public bool HasTitle { get; set; }
        public string Title { get; set; }
        public int Index { get; set; }
        public int PageId { get; set; }
        public TextContent TextContent { get; set; }
        public CalendarContent CalendarContent { get; set; }
        public TabContent TabContent { get; set; }
        public FileContent FileContent { get; set; }
        public Page Page { get; set; }
    }

    public class TextContent
    {
        public int Id { get; set; }
        public string Content { get; set; }
    }

    public class CalendarContent
    {
        public int Id { get; set; }
        public int CalendarId { get; set; }
        public Calendar Calendar { get; set; }
    }

    public class FileContent
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string FileType { get; set; }
        public List<File> Files { get; set; }
    }

    public class TabContent
    {
        public int Id { get; set; }
        public List<Tab> Tabs { get; set; }   
    }

    public class Tab
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int Index { get; set; }
        public int TabContentId { get; set; }
        public TabContent TabContent { get; set; }
        public TextContent TextContent { get; set; }
    }

    public class TextContentBackup
    {
        public int Id { get; set; }
        public string Content { get; set; }
        public DateTime CreationDate { get; set; }
        public int TextContentId { get; set; }
        public TextContent TextContent { get; set; }
    }
}